app/models/user.rb

```diff
@@ -1510,4 +1515,19 @@ def anonymize(new_wca_id = nil)
       country_iso2: special_account? ? country_iso2 : nil,
     )
   end
+
+  def transfer_data_to(user)
+    competition_organizers.update_all(organizer_id: user.id)
+    competition_delegates.update_all(delegate_id: user.id)
+    competitions_results_posted.update_all(results_posted_by: user.id)
+    competitions_announced.update_all(announced_by: user.id)
+    roles.update_all(user_id: user.id)
+    registrations.update_all(user_id: user.id)
+
+    return if wca_id.blank?
+
+    wca_id_to_be_transferred = wca_id
+    self.update!(wca_id: nil) # Must remove WCA ID before adding it as it is unique in the Users table.
```

Comment:
Can this backfire with validations? I think a lot of validations that we have depend on "Does this person have a WCA ID?"

And seeing that you already transferred a lot of relevant information (roles etc.) above, some of the validations might "freak out" if you're suddenly trying to nullify a WCA ID

Reply:
I don't think that will be the case, because this was already happening in reassign WCA ID script. Also, will be going through few rounds of testing before the final merging. So if there are any issues, we will catch there.

Reply:
Your argument doesn't hold because the old script was doing ` User.where(id: account1_user.id).update_all(wca_id: nil)`. This directly translates to a raw SQL query, which Rails executes without performing model validation checks.
Your new code runs directly on an instance of User using `update!`, which means that validations are run and errors are thrown if validations fail. This was not the case in the past implementation.

More reading: https://www.mintbit.com/blog/update-all-vs-update-in-rails/

Reply:
I tested locally by merging two accounts, and one of them was of Sébastien, who have lots of data in website, and it got merged successfully. So I assume it won't backfire validations.

Reply:
Are you confident that we have enough tests in place, so that future validation conflicts will be detected early on?